[BUG] 8.1.0 fails to compile due to deprecated calloc

Sorry if I fail to see something obvious, but neither the changelog nor the valkey code gave me an idea if calloc is marked as deprecated intentionally here. Otoh I wrote my last lines of C code years ago, so it's quite possible I overlook something. I'd appreciate any hint :)

**Describe the bug**

Fails to compile under Arch Linux while 8.0.2 compiles fine:

```
    CC cli_common.o
    CC cli_commands.o
    CC valkey-benchmark.o
allocator_defrag.c: In function ‘allocatorDefragInit’:
allocator_defrag.c:282:5: error: ‘calloc’ is deprecated [-Werror=deprecated-declarations]
  282 |     je_cb.bin_info = je_calloc(je_cb.nbins, sizeof(jeBinInfo));
      |     ^~~~~
In file included from zmalloc.h:100,
                 from server.h:73,
                 from allocator_defrag.c:46:
/usr/include/malloc.h:43:14: note: declared here
   43 | extern void *calloc (size_t __nmemb, size_t __size)
      |              ^~~~~~
allocator_defrag.c:284:5: error: ‘calloc’ is deprecated [-Werror=deprecated-declarations]
  284 |     je_usage_info = je_calloc(je_cb.nbins, sizeof(jemallocBinUsageData));
      |     ^~~~~~~~~~~~~
/usr/include/malloc.h:43:14: note: declared here
   43 | extern void *calloc (size_t __nmemb, size_t __size)
      |              ^~~~~~
cc1: some warnings being treated as errors
make[1]: *** [Makefile:538: allocator_defrag.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/build/valkey/src/valkey-8.1.0/src'
make: *** [Makefile:6: all] Error 2
make: Leaving directory '/build/valkey/src/valkey-8.1.0'
```

**To reproduce**

Compile valkey 8.1.0 under Arch Linux.

**Expected behavior**

Compiles fine, as 8.0.2 does.

**Additional information**

Any additional information that is relevant to the problem.

--- a/src/allocator_defrag.c
+++ b/src/allocator_defrag.c
@@ -279,9 +279,9 @@ int allocatorDefragInit(void) {
     je_res = je_mallctl("arenas.nbins", &je_cb.nbins, &sz, NULL, 0);
     assert(je_res == 0 && je_cb.nbins != 0);
 
-    je_cb.bin_info = je_calloc(je_cb.nbins, sizeof(jeBinInfo));
+    je_cb.bin_info = zcalloc_num(je_cb.nbins, sizeof(jeBinInfo));
     assert(je_cb.bin_info != NULL);
-    je_usage_info = je_calloc(je_cb.nbins, sizeof(jemallocBinUsageData));
+    je_usage_info = zcalloc_num(je_cb.nbins, sizeof(jemallocBinUsageData));
     assert(je_usage_info != NULL);
 
     for (unsigned j = 0; j < je_cb.nbins; j++) {

